custom_components/ramses_cc/config_flow.py

```diff
@@ -0,0 +1,591 @@
+"""Config flow to configure Ramses integration."""
+
+from abc import abstractmethod
+from copy import deepcopy
+import logging
+import re
+from typing import Any
+
+from ramses_rf.schemas import SCH_GATEWAY_DICT, SCH_GLOBAL_SCHEMAS, SZ_SCHEMA
+from ramses_tx.schemas import (
+    SCH_ENGINE_DICT,
+    SCH_SERIAL_PORT_CONFIG,
+    SZ_ENFORCE_KNOWN_LIST,
+    SZ_FILE_NAME,
+    SZ_KNOWN_LIST,
+    SZ_PACKET_LOG,
+    SZ_PORT_NAME,
+    SZ_ROTATE_BACKUPS,
+    SZ_ROTATE_BYTES,
+    SZ_SERIAL_PORT,
+)
+from serial.tools import list_ports  # type: ignore[import-untyped]
+import voluptuous as vol  # type: ignore[import-untyped]
+
+from homeassistant.components import usb
+from homeassistant.config_entries import (
+    ConfigEntry,
+    ConfigEntryState,
+    ConfigFlow,
+    OptionsFlow,
+)
+from homeassistant.const import CONF_SCAN_INTERVAL
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.data_entry_flow import FlowHandler, FlowResult
+from homeassistant.helpers import config_validation as cv, selector
+from homeassistant.helpers.storage import Store
+
+from .const import (
+    CONF_ADVANCED_FEATURES,
+    CONF_MESSAGE_EVENTS,
+    CONF_RAMSES_RF,
+    CONF_SCHEMA,
+    CONF_SEND_PACKET,
+    DOMAIN,
+    STORAGE_KEY,
+    STORAGE_VERSION,
+    SZ_CLIENT_STATE,
+    SZ_PACKETS,
+)
+from .schemas import SCH_GLOBAL_TRAITS_DICT
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_MANUAL_PATH = "Enter Manually"
+
+
+def get_usb_ports() -> dict[str, str]:
+    """Return a dict of USB ports and their friendly names."""
```

Comment:
As an enhancement it would be good to sort these in order of likelihood of being the correct device. The UI will later default to the first item in the list on initial entry creation.

Reply:
> As an enhancement it would be good to sort these in order of likelihood of being the correct device. The UI will later default to the first item in the list on initial entry creation.

I can't think of any way, except via its USB descriptor fields.

Or you could ping it, and see if it behaves appropriately - how could you do that, if another library is using it.

Reply:
Yeah, I was just thinking of sorting based on the existence of, say, "evofw3" in the description.

Reply:
There's a new dongle coming... might be very popular... will do:
 - ramses_rf <---> MQTT (over WiFi) <---> dongle <---> RF

It's by-id is: `/dev/serial/by-id/usb-Espressif_USB_JTAG_serial_debug_unit_84:FC:E6:64:45:8C-if00`

Reply:
That does sound interesting. I think I'll leave this alone for the time being. It seems like the evofw3 is likely to be the only device that's easy to detect reliably and the extra complexity the sorting adds is perhaps not really worth it just to get it to show up first in the list.